src/Hypergrid.js

```diff
@@ -3246,10 +3243,18 @@ Hypergrid.prototype = {
     newRectangle: function(x, y, width, height) {
         return new Rectangle(x, y, width, height);
     },
-    getFormattedValue: function(x, y) {
+
+    /**
+     * @param {number} c - grid cell coordinate
+     * @param {number} r - grid cell coordinate
+     * @returns {*}
+     */
+    getFormattedValue: function(c, r) {
+        var x = this.behavior.getActiveColumn(c),
+            y = r - this.getHeaderRowCount();
         return this.formatValue(
-            this.getColumnProperties(x).format,
-            this.getValue(x, y + this.getHeaderRowCount())
+            this.behavior.getCellProperty(x, y, 'format'),
```

Comment:
Access the cell props obj _when it exists_ (and has column props obj as prototype); _or_ the column props obj when cell props obj does not exist.